src/__tests__/local-state/general.ts

```diff
@@ -1196,4 +1196,45 @@ describe('Combining client and server state/operations', () => {
       },
     });
   });
+
+  itAsync('should handle server errors when field is required', (resolve, reject) => {
```

Comment:
I'm not quite sold on the description here. While I get the reason behind this, I'd like to be a bit more specific about when this situation arises. 

I'm understanding the exception to be thrown when the root `data` property returns null. In cases where partial data is returned, local resolvers work as expected. Perhaps we could update this to something like:

```suggestion
  itAsync('handles server errors when root data property is null', (resolve, reject) => {
```

Feel free to wordsmith this a bit more, but I think something like this better captures the scenario we are testing. There are a variety of reasons the server might return `null` for the root `data` field, not just because a field is non-null, so I'd love for the test to reflect this.